### ISSUE TITLE
factory-4.1.0 does not respect the CC Variable given to configure

### ISSUE THREAD

**Issue by tom111** [Macaulay2/M2#645]
On MacOS I configured with:

    CC=gcc-6 CXX=g++-6 CPP=cpp-6 LD=gcc-6 ../../configure --enable-download

(which enabled gcc-6 from homebrew), but it seems that factory (and potentially also other included libs) are calling gcc directly, such that they are compiled with clang.

**Comment by tom111** [comment]
Then further down in the compile process I get a segmentation fault: 
g++-6 -std=gnu++11 -O2 -pipe -Wall -Wno-shadow -Wcast-qual -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -Wconversion  -Wno-error=pragmas -std=gnu++11 -O2 -fabi-version=6 -I/Users/tom/temp/M2/M2/BUILD/tom/usr-host/include -O2 -Wall -g -std=gnu++11 -O2 -fabi-version=6 -fopenmp -std=gnu++11 -O2 -fabi-version=6 -I/Users/tom/temp/M2/M2/BUILD/tom/usr-host/include -Wreturn-type -I../../../../Macaulay2/d -I../../../../Macaulay2/d/../e -I../../../../Macaulay2/d/../c -I../../../../Macaulay2/d/../regex -isystem /Users/tom/temp/M2/M2/BUILD/tom/../../include -isystem /Users/tom/temp/M2/M2/BUILD/tom/include -isystem /Users/tom/temp/M2/M2/BUILD/tom/usr-host/include  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/memtailor/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/mathic/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/mathicgb/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/fflas_ffpack/src -isystem /usr/include/libxml2 -isystem /usr/local/Cellar/bdw-gc/7.6.0/include -isystem /usr/local/Cellar/libatomic_ops/7.4.6/include  -DNDEBUG -DNDEBUG -Wno-fatal-errors  -c -Wno-old-style-cast -o M2-factory.o ../../../../Macaulay2/d/M2-factory.cpp
CC M2types.c
make[1]: *** [all-in-d] Segmentation fault: 11
make[1]: Leaving directory `/Users/tom/temp/M2/M2/BUILD/tom/Macaulay2'
make: *** [all-in-Macaulay2] Error 2
make: Leaving directory `/Users/tom/temp/M2/M2/BUILD/tom'

I don't remember, do I have to use gcc-4.9 to compile M2?

**Comment by tom111** [comment]
With 

`CC=gcc-4.9 CXX=g++-4.9 CPP=cpp-4.9 LD=gcc-4.9 ../../configure --enable-download`

and

`CC=gcc-4.9 CXX=g++-4.9 CPP=cpp-4.9 LD=gcc-4.9 make`

I end up here:

```
 g++-4.9 -std=gnu++11 -O2 -pipe -Wall -Wno-shadow -Wcast-qual -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -Wconversion  -Wno-error=pragmas -std=gnu++11 -O2 -fabi-version=6 -I/Users/tom/temp/M2/M2/BUILD/tom/usr-host/include -O2 -Wall -g -std=gnu++11 -O2 -fabi-version=6 -fopenmp -std=gnu++11 -O2 -fabi-version=6 -I/Users/tom/temp/M2/M2/BUILD/tom/usr-host/include -Wreturn-type -I../../../../Macaulay2/d -I../../../../Macaulay2/d/../e -I../../../../Macaulay2/d/../c -I../../../../Macaulay2/d/../regex -isystem /Users/tom/temp/M2/M2/BUILD/tom/../../include -isystem /Users/tom/temp/M2/M2/BUILD/tom/include -isystem /Users/tom/temp/M2/M2/BUILD/tom/usr-host/include  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/memtailor/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/mathic/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/mathicgb/src  -I/Users/tom/temp/M2/M2/BUILD/tom/../../submodules/fflas_ffpack/src -isystem /usr/include/libxml2 -isystem /usr/local/Cellar/bdw-gc/7.6.0/include -isystem /usr/local/Cellar/libatomic_ops/7.4.6/include  -DNDEBUG -DNDEBUG -Wno-fatal-errors  -c -Wno-old-style-cast -o M2-factory.o ../../../../Macaulay2/d/M2-factory.cpp
CC M2types.c
CC scclib.c
../../../../Macaulay2/d/scclib.c:5:26: fatal error: expr-exports.h: No such file or directory
 #include "expr-exports.h"
                          ^
compilation terminated.
make[2]: *** [scclib.o] Error 1
make[2]: Leaving directory `/Users/tom/temp/M2/M2/BUILD/tom/Macaulay2/d'
make[1]: *** [all-in-d] Error 2
make[1]: Leaving directory `/Users/tom/temp/M2/M2/BUILD/tom/Macaulay2'
make: *** [all-in-Macaulay2] Error 2
make: Leaving directory `/Users/tom/temp/M2/M2/BUILD/tom'
```

**Comment by DanGrayson** [comment]
No, now M2 seems to compile with all modern compilers.  Under Mac OS X we now regularly compile with clang ( /usr/bin/gcc ).

**Comment by DanGrayson** [comment]
I'll try it out on the Mac with gcc-6.

**Comment by DanGrayson** [comment]
In particular, we are now able to use the native compiler on all systems.

**Comment by DanGrayson** [comment]
By the way, on Mac OS, you may simply use the script INSTALL.mac.

**Comment by DanGrayson** [comment]
Now I recall that we saw this:
```
CC M2types.c
make[1]: *** [all-in-d] Segmentation fault: 11
```
while we were preparing the release, and we had no way to "fix" it.  But now I have a little bit of time so can take a closer look, if it happens here with gcc-6.

**Comment by DanGrayson** [comment]
Nope, gcc-6 under Mac OS X compiles M2types.c just fine, for me.

**Comment by DanGrayson** [comment]
Re: `fatal error: expr-exports.h: No such file or directory`

The work-around to avoid that error is to use a more modern version of `make`, whose 3.82 vintage under Mac OS X has a bug.  I like `gmake`, as installed by `brew install make`.

**Comment by DanGrayson** [comment]
Re: ` it seems that factory (and potentially also other included libs) are calling gcc directly, such that they are compiled with clang. `

Not here, factory obeys and compiles with gcc-6.
